### PR TITLE
Replace `got` w/ `node-fetch`

### LINKED ISSUE #109: Replace `got` with alternative that doesn't depend on `@types/node`
### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 

### Subject

I'm trying to build a [Remix}(https://github.com/remix-run/remix) application that is deployed to Cloudflare Pages. This is an environment that is not Node.js based. My application deploys successfully but `tsc` fails because the environment type declarations for the Cloudflare environment conflict with `@types/node`. But I don't even have `@types/node` installed. Or actually, I do. In my lockfile I find `@remix-run/dev` depends on `xdm`, which depends on `got`, which depends on `@types/cacheable-request`, which depends on `@types/node` ( which is cool, because `cacheable-request` is a Node.js library ). The two environment types ( `@cloudflare/workers-types` and `@types/node` ) conflict. 

### Problem

`xdm` depends on `got` and thus installs `@types/node`, which makes `tsc` fail in Cloudflare environments.

TODO

### Solution

I would like to ask whether it would be possible to swap out `got` for an alternative that does not depend on `@types/node` I'd be happy to implement this change. I've checked and [`got` is only used on one line in the project](https://github.com/wooorm/xdm/blob/dc1193879889ac65eb6754f683585f0d7f2b145a/lib/integration/esbuild.js#L100). It would be trivial to replace this with [`node-fetch-cache`](https://www.npmjs.com/package/node-fetch-cache), which does not depend on `@types/node`.

### Alternatives

There are alternative replacement libraries, but `node-fetch-cache` offers all the features we used from `got` and doesn't depend on `@types/node`.

### PR DESCRIPTION
**Pull request by jensmeindertsma**
This pull request fixes #109. It replaces the `got` package which is only used once, with an alternative ( `node-fetch`, which against intuitions does not depend on `@types/node` where `got` did ). 

This means that there are now not any issues when running `tsc` in a Cloudflare Workers project which uses `xdm`. This came up specifically while building a Remix application, as `@remix-run/dev` depends on `xdm`.

I made sure to replicate the existing caching set up, as that is not included in `node-fetch`.

Thanks for this awesome project 🙏

### CODE REVIEW

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
Can you rebase? I fixed the unrelated error on main

File: lib/integration/esbuild.js [hunk 1]
```diff
@@ -97,7 +97,17 @@ export function esbuild(options = {}) {
...
+      /** @type {string} */
+      let contents
+
+      const cachedContents = cache.get(href)
+      if (cachedContents) {
+        contents = cachedContents
+      } else {
+        contents = await (await fetch(href)).text()
+        cache.set(href, contents)
+      }
```
Comment (ChristianMurphy) [inline lib/integration/esbuild.js:110]: This issue may have already existed before, but now is more visible.
It's a bit worrying that the caching system doesn't have memory management (max cached items, max memory usable, etc).
On massive projects, caching could become an issue exceeding the memory node is allowed.

**Review by ChristianMurphy — APPROVED** [review]
thanks @jensmeindertsma!

File: lib/integration/esbuild.js (hunk 1 above)
Comment (jensmeindertsma) [reply lib/integration/esbuild.js:110]: Yeah, not sure whether that should be improved as part of this PR though, might be better do to it in a separate PR?

Comment (wooorm) [reply lib/integration/esbuild.js:110]: I’m not *too* worried about it. Could use something smarter in the future (an LRU cache?), but this is probably fine for now in build scripts

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
Released, thanks!

**Pull request merged**